tests/functional/openstackversion_controller_test.go

```diff
@@ -0,0 +1,141 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package functional_test
+
+import (
+	"os"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	k8s_corev1 "k8s.io/api/core/v1"
+
+	. "github.com/openstack-k8s-operators/lib-common/modules/common/test/helpers"
+	corev1 "github.com/openstack-k8s-operators/openstack-operator/apis/core/v1beta1"
+)
+
+var _ = Describe("OpenStackOperator controller", func() {
+	BeforeEach(func() {
+		// lib-common uses OPERATOR_TEMPLATES env var to locate the "templates"
+		// directory of the operator. We need to set them othervise lib-common
+		// will fail to generate the ConfigMap as it does not find common.sh
+		err := os.Setenv("OPERATOR_TEMPLATES", "../../templates")
+		Expect(err).NotTo(HaveOccurred())
+
+	})
+
+	When("A default OpenStackVersion instance is created", func() {
+		BeforeEach(func() {
+			DeferCleanup(
+				th.DeleteInstance,
+				CreateOpenStackVersion(names.OpenStackVersionName, GetDefaultOpenStackVersionSpec()),
+			)
+		})
+		It("should initialize container images", func() {
+			Eventually(func(g Gomega) {
+
+				th.ExpectCondition(
+					names.OpenStackVersionName,
+					ConditionGetterFunc(OpenStackVersionConditionGetter),
+					corev1.OpenStackVersionInitialized,
+					k8s_corev1.ConditionTrue,
+				)
+
+				version := GetOpenStackVersion(names.OpenStackVersionName)
+				g.Expect(version).Should(Not(BeNil()))
+
+				g.Expect(*version.Status.AvailableVersion).Should(Equal("0.0.1"))
+				g.Expect(version.Spec.TargetVersion).Should(Equal("0.0.1"))
+
+				g.Expect(*version.Status.ContainerImages.AgentImage).ShouldNot(BeNil())
```

Comment:
here and below: why do you take the address of the image field? It is already a *string.

Reply:
initially I was checking the empty string values. Since I changed to just checking for nil I can remove the dereference bits I think